kazoo/tests/test_client.py

```diff
@@ -650,140 +677,148 @@ def test_create_makepath_incompatible_acls(self):
         client = self.client
         path, stat1 = client.create("/1", b"bytes", include_data=True)
         data, stat2 = client.get("/1")
-        eq_(data, b"bytes")
-        eq_(stat1, stat2)
+        assert data == b"bytes"
+        assert stat1 == stat2
 
     def test_create_get_set(self):
         nodepath = "/" + uuid.uuid4().hex
 
         self.client.create(nodepath, b"sandwich", ephemeral=True)
 
         data, stat = self.client.get(nodepath)
-        eq_(data, b"sandwich")
+        assert data == b"sandwich"
 
         newstat = self.client.set(nodepath, b"hats", stat.version)
-        self.assertTrue(newstat)
+        assert newstat
         assert newstat.version > stat.version
 
         # Some other checks of the ZnodeStat object we got
-        eq_(newstat.acl_version, stat.acl_version)
-        eq_(newstat.created, stat.ctime / 1000.0)
-        eq_(newstat.last_modified, newstat.mtime / 1000.0)
-        eq_(newstat.owner_session_id, stat.ephemeralOwner)
-        eq_(newstat.creation_transaction_id, stat.czxid)
-        eq_(newstat.last_modified_transaction_id, newstat.mzxid)
-        eq_(newstat.data_length, newstat.dataLength)
-        eq_(newstat.children_count, stat.numChildren)
-        eq_(newstat.children_version, stat.cversion)
+        assert newstat.acl_version == stat.acl_version
+        assert newstat.created == stat.ctime / 1000.0
+        assert newstat.last_modified == newstat.mtime / 1000.0
+        assert newstat.owner_session_id == stat.ephemeralOwner
+        assert newstat.creation_transaction_id == stat.czxid
+        assert newstat.last_modified_transaction_id == newstat.mzxid
+        assert newstat.data_length == newstat.dataLength
+        assert newstat.children_count == stat.numChildren
+        assert newstat.children_version == stat.cversion
 
     def test_get_invalid_arguments(self):
         client = self.client
-        self.assertRaises(TypeError, client.get, ('a', 'b'))
-        self.assertRaises(TypeError, client.get, 'a', watch=True)
+        with pytest.raises(TypeError):
+            client.get(('a', 'b'))
+        with pytest.raises(TypeError):
+            client.get('a', watch=True)
 
     def test_bad_argument(self):
         client = self.client
         client.ensure_path("/1")
-        self.assertRaises(TypeError, self.client.set, "/1", 1)
+        with pytest.raises(TypeError):
+            self.client.set("/1", 1)
 
     def test_ensure_path(self):
         client = self.client
         client.ensure_path("/1/2")
-        self.assertTrue(client.exists("/1/2"))
+        assert client.exists("/1/2")
 
         client.ensure_path("/1/2/3/4")
-        self.assertTrue(client.exists("/1/2/3/4"))
+        assert client.exists("/1/2/3/4")
 
+    @pytest.mark.skip("BUG: sync() call is not chroot'd")
     def test_sync(self):
         client = self.client
-        self.assertTrue(client.sync('/'), '/')
+        assert client.sync('/') == '/'
```

Comment:
The origin test was broken.  Turns out `sync()` is not unchrooted.

Reply:
I am not sure to understand this one, what is the problem with this test?

Reply:
Sync return the *unchrooted* path. So the test fails